### PR TITLE
Pool array builders so we produce less garbage.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by dpoeschl** [comment]
It'd be cool if we could test the efficacy of this sort of thing. /cc: @rchande

File: src/EditorFeatures/Test/Structure/AbstractSyntaxNodeStructureProviderTests.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public abstract class AbstractSyntaxNodeStructureProviderTests<TSyntaxNode> : Ab
             }
 
             var outliner = CreateProvider();
-            var actualRegions = ImmutableArray.CreateBuilder<BlockSpan>();
+            var actualRegions = ArrayBuilder<BlockSpan>.GetInstance();
```
Comment (dpoeschl) [inline src/EditorFeatures/Test/Structure/AbstractSyntaxNodeStructureProviderTests.cs]: Why doesn't this one need cleaned up?

**Comment by dpoeschl** [comment]
LGTM, but I still want to know why one (or more) of them didn't need to be free'd....

**Pull request merged**